spec/models/user_spec.rb

```diff
@@ -160,4 +162,98 @@
       end
     end
   end
+
+  describe '#visible_assessments' do
+
+    context "when there are no assessments" do
+      let(:new_user) {create :user, user_name: 'testusername', first_name: 'testfname',
+                               last_name: "testlname", type: "Student"}
+      it "should return an empty list" do
+        expect(new_user.visible_assessments()).to match_array([])
+      end
+    end
+    context "when there are assessments" do
+      before :each do
+        @new_section = create(:section)
```

Comment:
use `let` variables instead of creating instance variables in a before block

Reply:
for example: 

```ruby
let(:new_section) { create :section }
let(:assignment_visible) { create :assignment, due_date: 2.days.from_now, assignment_properties_attributes: { section_due_dates_type: true }  }

# etc. etc.
```

If you want to force the creation of each object before each test (ie. they're not lazily loaded) you should use the `let!` syntax (https://relishapp.com/rspec/rspec-core/v/2-5/docs/helper-methods/let-and-let)